diff.py

```diff
@@ -1115,7 +1124,8 @@ def run_make_capture_output(
 
 def restrict_to_function(dump: str, fn_name: str) -> str:
     try:
-        ind = dump.index("\n", dump.index(f"<{fn_name}>:"))
+        # Find the start of the line that contains "<fn_name>:"
+        ind = dump.rfind("\n", 0, dump.index(f"<{fn_name}>:"))
         return dump[ind + 1 :]
```

Comment:
```suggestion
        ind = dump.rfind("\n", 0, dump.index(f"<{fn_name}>:")) + 1
        return dump[ind:]
```

Pretty neat that this works even if the line is the very first one. It's not often -1 sentinels work in your favor like that.